### ISSUE TITLE
Driven Light Material Properties are not updating in linked library override projects.

### ISSUE THREAD

**Issue by dibli-goost** [BlenderNPR/BEER#529]
### Malt version

Node Groups

### Blender version

Blender 3.6.5

### OS

Windows 10

### Hardware info

AMD Ryzen Threadripper 1920x 12-Core Processor, 3950 Mhz, 24 Logical Processors | 32GB RAM | NVIDIA GeForce GTX 1080 Ti 

### Issue description and reproduction steps

Within the supplied blend are an Armature and a Sun light. The Sun light has a shader on it, within it's node tree is a float parameter, and it has a property override applied so that a driver can be placed upon it. This driver is connected to a property found on the bone in the armature. You can change the custom property value and see it reflected on the respective Sun parameter.

Link the two objects from the supplied blend into a new file. Perform a library override hierarchy on the two, and change the custom property, you'll see the change is not reflected on the Sun light. This might be because the material containing the shader isn't getting a library override. You can shift-click on the link and the following error will appear in the console:

`The data-block MASun - Shader could not be overridden`

I should note, that the only reason why I ran into this was because some of the light properties, such as Strength and Radius, cannot be controlled by drivers. I needed to build a light rig, so I needed those properties to be driven. I tried doing it through the light shader instead, and it seemed fine until I linked into a new project file.

### Attachments

[overridebug.zip](https://github.com/bnpr/Malt/files/13209656/overridebug.zip)

**Comment by pragma37** [comment]
> `The data-block MASun - Shader could not be overridden`

I have not looked in depth into it, but all Malt materials are marked as `LIBRARY_EDITABLE` and `LIBRARY_OVERRIDABLE`, so, I guess this is a limitation of the override system?

> I should note, that the only reason why I ran into this was because some of the light properties, such as Strength and Radius, cannot be controlled by drivers.

The Light properties were missing the `ANIMATABLE` tag, so at least that part is solved now. 🙂

**Comment by dibli-goost** [comment]
> I have not looked in depth into it, but all Malt materials are marked as `LIBRARY_EDITABLE` and `LIBRARY_OVERRIDABLE`, so, I guess this is a limitation of the override system?

Yeah, admittedly this might not be a problem with Malt, Overrides are very janky, so it might be something up with Blender itself. I personally haven't gotten too invested with light shaders quite yet, so it's not a high priority as long as the main properties can be driven. Thanks for the fix!
